yellowbrick/cluster/silhouette.py

```diff
@@ -97,13 +104,15 @@ class SilhouetteVisualizer(ClusteringScoreVisualizer):
     >>> model.poof()
     """
 
-    def __init__(self, model, ax=None, **kwargs):
+    def __init__(self, model, ax=None, colormap='set1', colors=None, **kwargs):
         super(SilhouetteVisualizer, self).__init__(model, ax=ax, **kwargs)
 
         # Visual Properties
-        # TODO: Fix the color handling
-        self.colormap = kwargs.get('colormap', 'set1')
-        self.color = kwargs.get('color', None)
+        # Use colors if it is given, otherwise attempt to find a colormap in
+        # styles.PALETTES. If not found, default to None. The colormap may
+        # yet still be found in resolve_colors
+        self.colormap = colormap
+        self.colors = colors if colors else PALETTES.get(colormap, None)
```

Comment:
change this to self.colors = kwargs.get('colors', None)

Ideally the user shouldn't provide both colormap and colors but if they do they will get a warning which is the expected behavior.

The reason you were getting a warning when you tested the updated code is because colormap was set to 'set1' and colors parameter was getting set using the colors in colormap.

Reply:
Thanks for reviewing @pdamodaran 
The reason I had changed the way `self.colors` was set was because I needed a way to get `PALETTES.get(colormap, None)` in there somewhere.

If we reach a point where `self.colormap = 'set1'` and `'self.colors = None`, when we reach [resolve_colors()](https://github.com/DistrictDataLabs/yellowbrick/blob/d2634e5899e3048cd423f814bdd6d141f5397ca9/yellowbrick/style/colors.py#L92) it will attempt `cm.get_cmap('set1')` and it will fail because `'set1'` does not exist. There does exist `'Set1'` with identical colors (I think) but the colors are in different orders and will break all tests.

The func `resolve_colors()` needs a way to be able to pull colormaps from `PALETTES`.